server/controllers/index.js

```diff
@@ -52,43 +54,65 @@ router.get('/auth/logout', auth, (req, res) => {
 });
 
 /* Create a room (a single instance of a game) */
-router.get('/api/room/create', auth, (req, res) => {
+router.post('/api/room/create', auth, (req, res) => {
     const { user } = req.session;
-    /* TODO: add check for when a room cannot be created and return an error */
     try {
-        let newGameState = new GameLogic(roomId);
-        let newGameRoom = {roomId, gameState: newGameState}
-        createRoom(user, roomId);
-        games.push(newGameRoom);
-        // Add the creator to the room
-        if (newGameState.canJoinGame(user)) { /* Game has not been started yet */
-            newGameState.joinGame(user, () => {
-                req.session.roomId = roomId;    
-                res.sendStatus(200);
-                roomId++;
-            });
-        } else { /* Game has already began */
-            res.status(400).json({message: "Game has already started."});
+        if (req.session.roomId !== undefined) {
+            const oldRoomId = req.session.roomId;
+            let game = getGameStateById(oldRoomId);
+            game.quitGame(user);
+            if (!game.getPlayers().length) games = games.filter(otherGame => otherGame !== game);
+            leaveRoom(user, oldRoomId);
+            req.session.roomId = undefined;
         }
+        let newGameState = new GameLogic(latestRoomId);
+        let newGameRoom = {roomId: latestRoomId, gameState: newGameState}
+        createRoom(user, latestRoomId, newGameState.getMinPlayers());
+        games.push(newGameRoom);
+        newGameState.joinGame(user);
+        req.session.roomId = latestRoomId;    
+        res.sendStatus(200);
+        latestRoomId++;
     } catch (err) {
         console.log(err);
         res.status(400).json({message: err.message});
     }
 });
 
+/* Join a room */
 router.post('/api/room/join', auth, (req, res) => {
     const { user } = req.session;
     const { roomId } = req.body;
     try {
-        joinRoom(user, roomId);
-        if (getGameStateById(roomId).canJoinGame(user)) {
-            getGameStateById(roomId).joinGame(user, () => {
-                req.session.roomId = roomId; 
-                res.sendStatus(200);
-            });
-        } else { /* Game has already began */
-            res.status(400).json({message: "Game has already started."});
+        if (req.session.roomId !== undefined) {
+            const oldRoomId = req.session.roomId;
+            let game = getGameStateById(oldRoomId);
+            game.quitGame(user);
+            if (!game.getPlayers().length) games = games.filter(otherGame => otherGame !== game);
+            leaveRoom(user, oldRoomId);
+            req.session.roomId = undefined;
```

Comment:
I don't usually like setting something to `undefined` myself. I'd prefer to use `null`. This is depends on you changing the `req.session.roomId !== undefined` line above.
```suggestion
            req.session.roomId = null;
```